test/framework/src/main/java/org/elasticsearch/test/ESTestCase.java

```diff
@@ -263,13 +264,35 @@ public void append(LogEvent event) {
             };
             leakAppender.start();
             Loggers.addAppender(leakLogger, leakAppender);
-            // shutdown hook so that when the test JVM exits, logging is shutdown too
-            Runtime.getRuntime().addShutdownHook(new Thread(() -> {
```

Comment:
This shouldn't have been put in the loop, the last two lines of this hook need only run once.